xcdat/regridder/accessor.py

```diff
@@ -80,11 +82,20 @@ def grid(self) -> xr.Dataset:
 
         return ds
 
-    def _get_axis_data(self, name: CFAxisName) -> Tuple[xr.DataArray, xr.DataArray]:
-        coord_var = get_axis_coord(self._ds, name)
+    def _get_axis_data(
+        self, name: CFAxisKey
+    ) -> Tuple[xr.DataArray, Optional[xr.DataArray]]:
+        coord_var = get_dim_coords(self._ds, name)
+
+        if isinstance(coord_var, xr.Dataset):
+            raise ValueError(
+                f"Multiple '{name}' axis dims were found in this dataset, "
+                f"{list(coord_var.dims)}. Please drop the unused dimension(s) before"
+                "getting grid information."
+            )
 
         try:
-            bounds_var = self._ds.bounds.get_bounds(name)
+            bounds_var = self._ds.bounds.get_bounds(name, coord_var.name)
         except KeyError:
             bounds_var = None
 
```

Comment:
xcdat grid APIs create a grids with a single dimension per axis (e.g., "lat" and "lon").

I updated `_get_axis_data()` to raise an error if an axis has multiple dimensions, which shouldn't be possible (this is for internal type checking and validation).